v0/destinations/heap/transform.js

```diff
@@ -39,20 +83,32 @@ function commonPayload(message, rawPayload, type) {
   const propertiesObj = {};
   let propsArray;
   let rudderPropertiesObj;
+  let email;
   switch (type) {
     case EventType.TRACK:
       propsArray = get(message, "properties")
         ? Object.keys(message.properties)
         : null;
       rudderPropertiesObj = message.properties;
-      rawPayload.identity = message.context.traits.email;
+      email =
+        (message.context.traits ? email : null) || message.properties.email;
+      if (email) rawPayload.identity = email;
+      else
+        throw Error("Email for identity is required for sending track calls.");
       break;
     case EventType.IDENTIFY:
       propsArray = get(message.context, "traits")
         ? Object.keys(message.context.traits)
         : null;
+      email =
+        message.context.traits.email ||
+        (message.properties ? message.properties.email : null);
```

Comment:
will identify have message.properties?

Reply:
properties can be sent through identify calls right? 
Thus checking once and then assigning.

Reply:
do we have that provision in our sdks?